optuna/integration/lightgbm_tuner/optimize.py

```diff
@@ -207,15 +210,15 @@ def __call__(self, trial):
             self.lgbm_params['num_leaves'] = trial.suggest_int(
                 'num_leaves', 2, 2 ** max_depth)
         if 'feature_fraction' in self.target_param_names:
-            param_value = trial.suggest_uniform('feature_fraction', 0.4, 1.0)
+            param_value = min(trial.suggest_uniform('feature_fraction', 0.4 - EPS, 1.0 + EPS), 1.0)
```

Comment:
I didn't understand why `- EPS` is required.
I'm not sure, but it seems that the right side expression should be replaced with `min(trial.suggest_uniform('feature_fraction', 0.4, 1.0 + EPS), 1.0)` or `max(min(trial.suggest_uniform('feature_fraction', 0.4 - EPS, 1.0 + EPS), 1.0), 0.4)`.
What do you think?

Reply:
You are right. `- EPS` for the lower bound is not necessary. To make the changes clear, I added comments related to the sample range. 280662c